### PR TITLE
🐳 Add initial automated memory test.

### PR DESCRIPTION
**Pull request by samccone**
This is the initial automated memory test for the menu component in material design light.

Running this file will check to see if any new DOM nodes or listeners
have been created, therefore automatically detecting memory leaks :potable_water: 

I am not sure what the CI setup is that this project is run on, however all that should be required is chrome driver in your path and chrome in the path, and things should just work :clap: 

Let me know if I can do anything else here :station: 

-------

This would have prevented #761 and will prevent it from regressing 

------

You can test this PR with :construction_worker: :muscle: 

`./node_modules/.bin/mocha test/memory/test.js --timeout=10000`

### CODE REVIEW

File: test/memory/test.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+function toggleMenu() {
+  driver.findElement(webdriver.By.css('#demo-menu-lower-left')).click();
+  driver.findElement(webdriver.By.css('#demo-menu-lower-left')).click();
+}
+
+console.log('Testing Menu Component');
+driver.get('file://' + path.join(__dirname, '../../dist/components/menu/demo.html'));
+toggleMenu();
+
+driver.sleep(1000);
```
Comment (samccone) [inline test/memory/test.js]: the leading and trailing sleeps are to let the initial and final animations finish :crossed_flags:

**Comment by Garbee** [comment]
Me likey.

**Comment by surma** [comment]
LGTM.

**Comment by samccone** [comment]
@surma any way we get get this running in your ci?

**Comment by surma** [comment]
The test doesn’t pass consistently for me (every 3rd or 4th run is failing) . Is that an issue with my system?

**Comment by samccone** [comment]
Hey @surma a few questions

What version of chrome? Which test(s) is/are failing?

---

That said I have seen several kinds of flakiness around these tests 
**xref** https://github.com/samccone/leaky/pull/3 That have yet to be able to quantify, or even figure out the cause. 

Thanks for taking a look :clap:

**Comment by surma** [comment]
Chrome 43.0.2357.134 (64-bit)
ChromeDriver 2.15.322455

There is only one test, so that’s the one failing :wink: 
```
 menu component
    1) does not leak


  0 passing (5s)
  1 failing

  1) menu component does not leak:

      no new nodes are allocated
      + expected - actual

      -65
      +228

      at Context.<anonymous> (test/memory/test.js:33:40)
      at Array.forEach (native)
```

**Comment by samccone** [comment]
oh sorry there are two asserts in the file, it looks like the node count assert is failing here, blah I must be missing something in my understanding of GC and nodes that can cause this race condition... Open to any ideas, I am going to continue my research :books:

**Comment by surma** [comment]
Thanks for your help, @samccone. Greatly appreciated :)

**Comment by addyosmani** [comment]
@samccone I know you've been busy with some other bugs lately :) Were you able to dive further into this?

**Comment by samccone** [comment]
This one is on the list for today's todos 😎🐴✈️⚡️

**Comment by addyosmani** [comment]
:rocket:

**Comment by samccone** [comment]
alright, just updated to the latest release of https://github.com/samccone/drool

I am fairly confident that there is a listener leak in the menu component per toggle action.

screen shot 2015-08-04 at 3 45 37 pm


The first run was 10 times
the second was 5 times
and the last was 2 times

Each growing linearly in listener counts. 

any thoughts @surma ?

**Comment by surma** [comment]
The test runs consistently now. YAY!

However, it doesn’t really fail in a way a CI could detect, does it? I’d like to run the test with `gulp test`, so I guess I’d like to wrap the test in a module. Does that make sense?

In the meanwhile, I’ll take a look at the menu code :)

**Comment by surma** [comment]
PS: Can’t reproduce the behavior you are seeing. Set `repeatCount` to 20, 10 and 5 resp.
screenshot 2015-08-05 11 08 20

**Comment by samccone** [comment]
huzzah consistency, :cry: your numbers vs mine. 


I just ran the tests again mac osx on top of the latest master.

`chromedriver 2.17.340128`

* chromium `<latest>`
* chrome canary `<latest>`
* chrome `<latest>`

I am able to reproduce my results each time. There has to be an inconsistency somewhere :tangerine:  Feel free to ping me off of this thread and we can isolate this difference so we can :ship: this.

**Comment by samccone** [comment]
:clap: @surma we got this! 

ok updated with an assert and such :dancer:

**Comment by surma** [comment]
Pressing the button⚠️

**Comment by samccone** [comment]
zomggggg :astonished:

**Pull request merged**